### ISSUE TITLE
Lint/DefEndAlignment, Lint/UselessComparison,Style/SpaceAroundOperators failing on super block

### ISSUE THREAD

**Issue by jgeiger** [bbatsov/rubocop#2784]
Seems there's an issue when parsing super with a block.

```shell
class Thing
  def send_command
    super { |response| }
  end
end
```

```shell
class Thing
  def send_command
    super do |response|
    end
  end
end
```

```shell
1 file inspected, no offenses detected

3 errors occurred:
An error occurred while Lint/DefEndAlignment cop was inspecting /app/app/models/thing.rb.
An error occurred while Lint/UselessComparison cop was inspecting /app/app/models/thing.rb.
An error occurred while Style/SpaceAroundOperators cop was inspecting /app/app/models/thing.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.37.0 (using Parser 2.3.0.2, running on ruby 2.2.4 x86_64-linux)
Finished in 0.1553650630012271 seconds
```

**Comment by turboladen** [comment]
I'm getting the same, plus some addition errors when used with `super` blocks (only happened after updating from 0.36 to 0.37).

```
An error occurred while Lint/DefEndAlignment cop was inspecting [redacted]
An error occurred while Lint/UselessComparison cop was inspecting [redacted]
An error occurred while Style/DotPosition cop was inspecting [redacted]
An error occurred while Style/IndentationWidth cop was inspecting [redacted]
An error occurred while Style/MultilineMethodCallIndentation cop was inspecting [redacted]
An error occurred while Style/MultilineOperationIndentation cop was inspecting [redacted]
An error occurred while Style/SpaceAroundOperators cop was inspecting [redacted]
An error occurred while Lint/DefEndAlignment cop was inspecting [redacted]
An error occurred while Lint/UselessComparison cop was inspecting [redacted]
An error occurred while Style/DotPosition cop was inspecting [redacted]
An error occurred while Style/IndentationWidth cop was inspecting [redacted]
An error occurred while Style/MultilineMethodCallIndentation cop was inspecting [redacted]
An error occurred while Style/MultilineOperationIndentation cop was inspecting [redacted]
An error occurred while Style/SpaceAroundOperators cop was inspecting [redacted]
```

**Comment by alexdowad** [comment]
Fixed in open PR.

**Comment by rrosenblum** [comment]
@turboladen, I am seeing the same thing as you. I expect that @alexdowad's PR, #2784, will fix the issues with the other cops as well
